hydra-node/src/Hydra/HeadLogic.hs

```diff
@@ -57,12 +49,26 @@ data ClientRequest tx
   | NewTx tx
   | Close
   | Contest
+  deriving (Generic)
 
 deriving instance Tx tx => Eq (ClientRequest tx)
 deriving instance Tx tx => Show (ClientRequest tx)
 deriving instance Tx tx => Read (ClientRequest tx)
 
-instance (ToJSON tx, ToJSON (UTxO tx)) => ToJSON (ClientRequest tx) where
+instance (Arbitrary tx, Arbitrary (UTxO tx)) => Arbitrary (ClientRequest tx) where
+  arbitrary = genericArbitrary
+
+  -- NOTE: Somehow, can't use 'genericShrink' here as GHC is complaining about
+  -- Overlapping instances with 'UTxO tx' even though for a fixed `tx`, there
+  -- should be only one 'UTxO tx'
```

Comment:
Maybe we're missing a functional dependency or so?

Reply:
Thought so. But really, it shouldn't be necessary here. I didn't give it much thought.